toolkit/tools/imagegen/installutils/overlay.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package installutils
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"microsoft.com/pkggen/imagegen/diskutils"
+	"microsoft.com/pkggen/internal/logger"
+)
+
+// Overlay Struct representing an overlay mount
+type Overlay struct {
```

Comment:
The `safechroot` package already contains an `NewOverlayMountPoint` routine that performs the lower/upper/workdir calculations and handles the mounting and unmounting. Is there a reason it can't be re-used instead of creating a new similair struct and routines?


https://github.com/microsoft/CBL-Mariner/blob/124daab644bcb9ed7ab65a94d44a5694b5ce569d/toolkit/tools/internal/safechroot/safechroot.go#L100

Reply:
My goal for this struct was to abstract names of the directories, creation and life time management from the caller. (Actually this was one of the review comments from the earlier review). With NewOverlayMountPoint the caller would need to do all these things.

---
In reply to: [573087014](https://github.com/microsoft/CBL-Mariner/pull/611#discussion_r573087014) [](ancestors = 573087014)